nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java

```diff
@@ -331,6 +331,9 @@ public class NiFiProperties extends ApplicationProperties {
     // performance tracking
     public static final String TRACK_PERFORMANCE_PERCENTAGE = "nifi.performance.tracking.percentage";
 
+    // flow action reporter
+    public static final String COMPONENT_FLOW_ACTION_REPORTER_IMPLEMENTATION = "nifi.components.flow.action.reporter.implementation";
```

Comment:
The `components` element can be removed. As this is a convenience property, it is probably best to avoid adding it to `NiFiProperties` and define it in the appropriate class.